packages/versioning/src/validate.ts

```diff
@@ -205,6 +210,74 @@ function validateMultiTypeReference(program: Program, source: Type) {
   }
 }
 
+/**
+ * Constructs a map of version to name for the the source.
+ */
+function getVersionedNameMap(
+  program: Program,
+  source: Type
+): Map<Version, string | undefined> | undefined {
+  const allVersions = getAllVersions(program, source);
+  if (allVersions === undefined) return undefined;
+
+  const map: Map<Version, string | undefined> = new Map(allVersions.map((v) => [v, undefined]));
+  const availMap = getAvailabilityMap(program, source);
```

Comment:
Should we compute that map only once and pass it to all the different validator, feels it is a cost that will add up

Reply:
Made that change, but it gets calculated per object (model/enum/union) so I'm not sure it saves a lot of computation.

Reply:
So I reverted the change due to the refactoring causing other tests that were passing to fail.

Reply:
You can get an availability map for any type (model, model property, enum, union variant, operation, etc.) so while there are multiple cases where we are building this map, each one is for a little specific piece. To be able to calculate "one map to rule them all" so that we would only need to calculate it once and share with all methods would be a major refactor.